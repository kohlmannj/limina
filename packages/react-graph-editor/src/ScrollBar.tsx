import React, { CSSProperties, SFC } from 'react';
import styled, { css, StyledComponent } from 'react-emotion';
import { OverflowMode, ScrollBarOrientation } from './index';

export interface IScrollBarProps {
  className?: string;
  orientation?: ScrollBarOrientation;
  overflow?: OverflowMode;
  progress?: number;
  scale?: number;
  style?: CSSProperties;
}

export interface IScrollBarTheme {
  thumbColor?: string;
  thumbWidth?: number;
  trackColor?: string;
  trackWidth?: number;
}

const getOvershootFactor = (progress = 0, min = 0, max = 1): number => {
  let overshoot;

  if (progress < min) {
    overshoot = 1 - Math.abs(progress);
  } else if (progress >= max) {
    overshoot = 1 - Math.abs(progress - max);
  } else {
    overshoot = 1;
  }

  return overshoot;
};

const positionForOrientation = (props: IStyledScrollBarProps) => css`
  left: ${!props.orientation || props.orientation === 'vertical' ? 'auto' : 0};
  right: ${!props.orientation || props.orientation === 'vertical'
    ? 0
    : `${
        props.theme && typeof props.theme.trackWidth === 'number' ? props.theme.trackWidth : 15
      }px`};
  top: ${!props.orientation || props.orientation === 'vertical' ? 0 : 'auto'};
  bottom: ${!props.orientation || props.orientation === 'vertical'
    ? `${props.theme && typeof props.theme.trackWidth === 'number' ? props.theme.trackWidth : 15}px`
    : 0};
`;

const Track = styled<IScrollBarProps, 'div'>('div')`
  background: ${props => props.theme.trackColor || `#efefef`};
  display: flex;
  flex-direction: ${props =>
    !props.orientation || props.orientation === 'vertical' ? 'column' : 'row'};
  align-items: center;
  justify-content: ${props =>
    typeof props.progress === 'number' && props.progress >= 0.5 ? 'flex-end' : 'flex-start'};
  ${props =>
    !props.orientation || props.orientation === 'vertical' ? 'width' : 'height'}: ${props =>
    `${typeof props.theme.trackWidth === 'number' ? props.theme.trackWidth : 15}px`};
  ${positionForOrientation};
`;

const ThumbContainer = styled<IScrollBarProps, 'div'>('div')`
  display: flex;
  flex-direction: inherit;
  justify-content: inherit;
  align-items: center;
  --overshoot-factor: ${props => getOvershootFactor(props.progress)};
  ${props => (!props.orientation || props.orientation === 'vertical' ? 'width' : 'height')}: 100%;
  /* Thumb Length */
  ${props =>
    !props.orientation || props.orientation === 'vertical' ? 'height' : 'width'}: ${props =>
    `calc(1 / ${props.scale || 1} * ${getOvershootFactor(props.progress)} * 100%)`};
  transform: ${props =>
    `translate${!props.orientation || props.orientation === 'vertical' ? 'Y' : 'X'}(${(Math.min(
      Math.max(props.progress || 0, 0),
      1
    ) +
      (typeof props.progress === 'number' && props.progress >= 0.5 ? -1 : 0)) *
    ((props.scale || 1) -
      1) /* *
      (typeof props.progress === 'number' && props.progress > 1
        ? (props.progress || 1) / getOvershootFactor(props.progress)
        : 1) */ *
      100}%)`};
`;

export interface IStyledScrollBarProps extends IScrollBarProps {
  theme?: IScrollBarTheme;
}

export const Thumb: StyledComponent<IScrollBarProps, IScrollBarTheme, {}> = styled<
  IScrollBarProps,
  'button'
>('button')`
  background: ${props => props.theme.thumbColor || '#666'};
  border-radius: ${props =>
    `${(typeof props.theme.thumbWidth === 'number' ? props.theme.thumbWidth : 15) / 2}px`};
  display: flex;
  padding: 0;
  border: 0;
  flex-direction: inherit;
  flex: 0 1 100%;
  margin: ${props =>
    `${!props.orientation || props.orientation === 'vertical' ? '' : '0 '}${((typeof props.theme
      .trackWidth === 'number'
      ? props.theme.trackWidth
      : 15) -
      (typeof props.theme.thumbWidth === 'number' ? props.theme.thumbWidth : 15)) /
      2}px${!props.orientation || props.orientation === 'vertical' ? ' 0' : ''}`};
  ${props =>
    !props.orientation || props.orientation === 'vertical' ? 'width' : 'height'}: ${props =>
    `${typeof props.theme.thumbWidth === 'number' ? props.theme.thumbWidth : 15}px`};
`;

export const ScrollBar: SFC<IScrollBarProps> = ({
  className,
  orientation,
  overflow,
  progress,
  scale,
  style,
}) => (
  <Track className={className} orientation={orientation} progress={progress} style={style}>
    <ThumbContainer orientation={orientation} progress={progress} scale={scale}>
      {typeof scale === 'number' &&
        (scale > 1 || (scale <= 1 && overflow === 'scroll')) && <Thumb orientation={orientation} />}
    </ThumbContainer>
  </Track>
);

ScrollBar.defaultProps = {
  orientation: 'vertical',
  overflow: 'auto',
  scale: 1,
};

export default ScrollBar;
