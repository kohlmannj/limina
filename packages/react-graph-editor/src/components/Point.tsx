/** @jsx jsx */
import { css, jsx, SerializedStyles } from '@emotion/core';
import { FunctionComponent } from 'react';
import styled, { StyledComponent } from '@emotion/styled';
import { ViewBoxContext } from './ViewBoxContext';
import { ThemeProps, ViewBox, OriginDirection, PointShape } from '..';

export interface PointProps {
  className?: string;
  selected?: boolean;
  x: number;
  y: number;
}

export interface StyledButtonProps extends PointProps, ViewBox {}

const getOriginTranslateFactor = (origin: OriginDirection) =>
  origin === 'left' || origin === 'top' ? -0.5 : 0.5;

export type DynamicStyle = (props: PointProps) => SerializedStyles | undefined;

export const shapes: Record<PointShape, DynamicStyle> = {
  circle: () => css`
    border-radius: 50%;
  `,
  diamond: () => undefined,
  square: () => undefined,
};

const selectedClassName = (props: PointProps & { theme?: ThemeProps }) => css`
  border-color: ${props.theme && props.theme.color ? props.theme.color : '#000'};
  background: #fff;
  box-shadow: 0 0 5px 2px rgba(59, 153, 252, 0.75);
`;

const translate = (props: StyledButtonProps) =>
  props.originX && props.originY
    ? `translate(${getOriginTranslateFactor(props.originX) * 100}%, ${getOriginTranslateFactor(
        props.originY
      ) * 100}%)`
    : '';

const position = (props: StyledButtonProps & { theme?: ThemeProps }) => css`
  position: absolute;
  ${props.originX && `${props.originX}: ${((props.x - props.minX) / props.width) * 100}%;`}
  ${props.originY && `${props.originY}: ${((props.y - props.minY) / props.height) * 100}%;`}
  transform: ${translate(props)} ${
  props.theme && props.theme.shape === 'diamond' ? 'rotate(45deg)' : ''
};
`;

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const Button: FunctionComponent<StyledButtonProps> = ({ className, selected, x, y, ...rest }) => (
  <button
    type="button"
    className={className}
    aria-label={`Point (${x}, ${y})`}
    title={`(${x}, ${y})`}
    {...rest}
  />
);

const StyledButton: StyledComponent<StyledButtonProps, StyledButtonProps, ThemeProps> = styled(
  Button
)`
  width: ${props => `${props.theme.thickness || 8}px`};
  height: ${props => `${props.theme.thickness || 8}px`};
  box-sizing: border-box;
  padding: 0;
  border: 1px solid transparent;
  background: ${props => props.theme.color || '#000'};
  box-shadow: 0 0 1px 1px rgba(0, 0, 0, 0.15);
  cursor: move;
  font-size: 0;
  line-height: 0;
  ${props => (!props.theme || props.theme.shape === 'circle') && `border-radius: 50%`};
  ${props => props.selected && selectedClassName};
  ${position};
  outline: 0;
  text-indent: 100%;
`;

export const Point: FunctionComponent<PointProps> = props => (
  <ViewBoxContext.Consumer>
    {(viewBox: ViewBox) => <StyledButton {...props} {...viewBox} />}
  </ViewBoxContext.Consumer>
);

Point.defaultProps = {
  selected: false,
};
