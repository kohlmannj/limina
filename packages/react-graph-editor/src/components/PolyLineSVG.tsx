import React, { FunctionComponent } from 'react';
import styled, { StyledComponent } from '@emotion/styled';
import { getDimensionsForPoints } from '../utils';
import { ThemeProps, PointTuple } from '..';

export type InterpolationKeyword = 'linear' | 'ease-in-out' | 'ease-in' | 'ease-out';

export type InterpolationFunction = (x: number) => number;

export type InterpolationKeywordOrFunction = InterpolationKeyword | InterpolationFunction;

export interface PolyLineSVGProps {
  className?: string;
  interpolation?: InterpolationKeywordOrFunction;
  label?: string;
  points: PointTuple[];
}

export const PolyLineSVG: FunctionComponent<PolyLineSVGProps> = ({
  className,
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  label,
  points,
  ...rest
}) => (
  <svg xmlns="http://www.w3.org/2000/svg" {...getDimensionsForPoints(points)}>
    <polyline
      className={className}
      vectorEffect="non-scaling-stroke"
      points={points.map(point => point.join(',')).join(' ')}
      {...rest}
    />
  </svg>
);

export const StyledPolyLineSVG: StyledComponent<
  PolyLineSVGProps,
  PolyLineSVGProps,
  ThemeProps
> = styled(PolyLineSVG)`
  fill: none;
  stroke: ${props => props.theme.color || '#000'};
  stroke-width: ${props => `${props.theme.thickness / 4 || 2}px`};
`;
