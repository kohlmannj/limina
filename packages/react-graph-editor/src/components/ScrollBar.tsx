import React, { CSSProperties, SFC } from 'react';
import styled, { css, StyledComponent } from 'react-emotion';
import { OverflowMode, ScrollBarOrientation } from '..';

export interface IScrollBarProps {
  className?: string;
  orientation?: ScrollBarOrientation;
  overflow?: OverflowMode;
  progress?: number;
  scale?: number;
  style?: CSSProperties;
}

export interface IScrollBarTheme {
  thumbColor?: string;
  thumbWidth?: number;
  trackColor?: string;
  trackWidth?: number;
}

const getOvershootFactor = (progress = 0, min = 0, max = 1): number => {
  let overshoot;

  if (progress < min) {
    overshoot = 1 - Math.abs(progress);
  } else if (progress >= max) {
    overshoot = 1 - Math.abs(progress - max);
  } else {
    overshoot = 1;
  }

  return overshoot;
};

const positionForOrientation = (props: IStyledScrollBarProps) => css`
  top: ${!props.orientation || props.orientation === 'vertical' ? 0 : 'auto'};
  right: ${!props.orientation || props.orientation === 'vertical'
    ? 0
    : `${
        props.theme && typeof props.theme.trackWidth === 'number' ? props.theme.trackWidth : 15
      }px`};
  bottom: ${!props.orientation || props.orientation === 'vertical'
    ? `${props.theme && typeof props.theme.trackWidth === 'number' ? props.theme.trackWidth : 15}px`
    : 0};
  left: ${!props.orientation || props.orientation === 'vertical' ? 'auto' : 0};
`;

const Track = styled<IScrollBarProps, 'div'>('div')`
  display: flex;
  flex-direction: ${props =>
    !props.orientation || props.orientation === 'vertical' ? 'column' : 'row'};
  align-items: center;
  justify-content: ${props =>
    typeof props.progress === 'number' && props.progress >= 0.5 ? 'flex-end' : 'flex-start'};
  background: ${props => props.theme.trackColor || `#efefef`};
  ${props =>
    !props.orientation || props.orientation === 'vertical' ? 'width' : 'height'}: ${props =>
    `${typeof props.theme.trackWidth === 'number' ? props.theme.trackWidth : 15}px`};
  ${positionForOrientation};
`;

const ThumbContainer = styled<IScrollBarProps, 'div'>('div')`
  display: flex;
  flex-direction: inherit;
  align-items: center;
  justify-content: inherit;
  ${props =>
    !props.orientation || props.orientation === 'vertical' ? 'width: 100%' : 'height: 100%'};
  ${props =>
    !props.orientation || props.orientation === 'vertical' ? 'height' : 'width'}: ${props =>
    `calc(1 / ${props.scale || 1} * ${getOvershootFactor(props.progress)} * 100%)`};
  transform: ${props =>
    `translate${!props.orientation || props.orientation === 'vertical' ? 'Y' : 'X'}(${(Math.min(
      Math.max(props.progress || 0, 0),
      1
    ) +
      (typeof props.progress === 'number' && props.progress >= 0.5 ? -1 : 0)) *
      ((props.scale || 1) - 1) *
      100}%)`};
`;

export interface IStyledScrollBarProps extends IScrollBarProps {
  theme?: IScrollBarTheme;
}

export const Thumb = styled<IScrollBarProps, 'button'>('button')`
  display: flex;
  flex: 0 1 100%;
  flex-direction: inherit;
  padding: 0;
  border: 0;
  margin: ${props =>
    `${!props.orientation || props.orientation === 'vertical' ? '' : '0 '}${((typeof props.theme
      .trackWidth === 'number'
      ? props.theme.trackWidth
      : 15) -
      (typeof props.theme.thumbWidth === 'number' ? props.theme.thumbWidth : 15)) /
      2}px${!props.orientation || props.orientation === 'vertical' ? ' 0' : ''}`};
  background: ${props => props.theme.thumbColor || '#666'};
  border-radius: ${props =>
    `${(typeof props.theme.thumbWidth === 'number' ? props.theme.thumbWidth : 15) / 2}px`};
  ${props =>
    !props.orientation || props.orientation === 'vertical' ? 'width' : 'height'}: ${props =>
    `${typeof props.theme.thumbWidth === 'number' ? props.theme.thumbWidth : 15}px`};
`;

export const ScrollBar: SFC<IScrollBarProps> = ({
  className,
  orientation,
  overflow,
  progress,
  scale,
  style,
}) => (
  <Track className={className} orientation={orientation} progress={progress} style={style}>
    <ThumbContainer orientation={orientation} progress={progress} scale={scale}>
      {typeof scale === 'number' &&
        (scale > 1 || (scale <= 1 && overflow === 'scroll')) && <Thumb orientation={orientation} />}
    </ThumbContainer>
  </Track>
);

ScrollBar.defaultProps = {
  orientation: 'vertical',
  overflow: 'auto',
  scale: 1,
};

export default ScrollBar;
