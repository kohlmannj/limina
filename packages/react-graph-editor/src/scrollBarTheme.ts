export interface ScrollBarTheme {
  thumbClassName?: string;
  thumbColor?: string;
  thumbWidth?: number;
  trackClassName?: string;
  trackColor?: string;
  trackWidth?: number;
}

export const defaults = {
  thumbColor: '#ccc',
  thumbWidth: 12,
  trackColor: '#333',
  trackWidth: 16,
};
