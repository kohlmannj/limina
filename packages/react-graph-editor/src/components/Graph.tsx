/** @jsx jsx */
import { jsx } from '@emotion/core';
import chroma from 'chroma-js';
import { ThemeProvider } from 'emotion-theming';
import { CSSProperties, Fragment, FunctionComponent } from 'react';
import styled, { StyledComponent } from '@emotion/styled';
import { getViewBoxForPoints } from '../utils';
import { Point, shapes } from './Point';
import { PolyLineSVGProps } from './PolyLineSVG';
import { ViewBoxContext } from './ViewBoxContext';
import { ThemeProps, OriginDirection, PointTuple } from '..';

const shapeNames = Object.keys(shapes);

export interface GraphProps {
  className?: string;
  lines: PolyLineSVGProps[];
  minX?: number;
  minY?: number;
  maxX?: number;
  maxY?: number;
  originX?: OriginDirection;
  originY?: OriginDirection;
  style?: CSSProperties;
}

const Graph: FunctionComponent<GraphProps> = ({
  className,
  lines,
  minX: minXProp,
  minY: minYProp,
  maxX: maxXProp,
  maxY: maxYProp,
  originX,
  originY,
  ...rest
}) => {
  const autoViewBox = getViewBoxForPoints(
    lines.reduce((reducedPoints: PointTuple[], { points }) => [...reducedPoints, ...points], [])
  );

  const minX = typeof minXProp === 'number' ? minXProp : autoViewBox.minX;
  const minY = typeof minYProp === 'number' ? minYProp : autoViewBox.minY;
  const maxX = typeof maxXProp === 'number' ? maxXProp : autoViewBox.maxX;
  const maxY = typeof maxYProp === 'number' ? maxYProp : autoViewBox.maxY;

  const viewBox = {
    minX,
    minY,
    maxX,
    maxY,
    originX,
    originY,
    width: typeof minX === 'number' && typeof maxX === 'number' ? Math.abs(maxX - minX) : 0,
    height: typeof minY === 'number' && typeof maxY === 'number' ? Math.abs(maxY - minY) : 0,
  };

  return (
    <div className={className} {...rest}>
      <ViewBoxContext.Provider value={viewBox}>
        {lines.map(({ label, points }, index) => (
          <ThemeProvider
            key={label}
            theme={{
              color: chroma.brewer.Set1[index % chroma.brewer.Set1.length],
              shape: shapeNames[index % shapeNames.length],
            }}
          >
            <Fragment>
              {points.map(([x, y]) => (
                <Point key={`${label}_${x}_${y}`} x={x} y={y} />
              ))}
            </Fragment>
          </ThemeProvider>
        ))}
      </ViewBoxContext.Provider>
    </div>
  );
};

Graph.defaultProps = {
  originX: 'left',
  originY: 'bottom',
};

const StyledGraph: StyledComponent<GraphProps, GraphProps, ThemeProps> = styled(Graph)`
  position: relative;
  width: 100%;
  height: 100%;
`;

const GraphContainer: FunctionComponent<GraphProps> = ({ className, style, ...rest }) => (
  <div className={className} style={style}>
    <StyledGraph {...rest} />
  </div>
);

export const StyledGraphContainer = styled(GraphContainer)`
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  padding: ${props => `${(props.theme.thickness || 8) / 2}px`};
`;
