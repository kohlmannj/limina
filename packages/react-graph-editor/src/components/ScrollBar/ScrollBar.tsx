/** @jsx jsx */
import { jsx } from '@emotion/core';
import { CSSProperties, FunctionComponent } from 'react';
import { ScrollBarTheme } from '../../scrollBarTheme';
import { ThumbContainer } from './components/ThumbContainer';
import { ThumbSegment, StyledThumbSegmentProps } from './components/ThumbSegment';
import { Track } from './components/Track';
import { OverflowMode, ScrollBarAxis } from '../..';

export interface ScrollBarProps {
  axis?: ScrollBarAxis;
  className?: string;
  overflow?: OverflowMode;
  progress?: number;
  scale?: number;
  style?: CSSProperties;
  thumbEndProps?: Partial<StyledThumbSegmentProps>;
  thumbMiddleProps?: Partial<StyledThumbSegmentProps>;
  thumbStartProps?: Partial<StyledThumbSegmentProps>;
}

export interface StyledScrollBarProps extends ScrollBarProps {
  theme?: Partial<ScrollBarTheme>;
}

export const ScrollBar: FunctionComponent<ScrollBarProps> = ({
  axis = 'y',
  className,
  overflow = 'scroll',
  progress,
  scale = 1,
  style,
  thumbEndProps,
  thumbMiddleProps,
  thumbStartProps,
}) => (
  <Track className={className} axis={axis} progress={progress} style={style}>
    {typeof scale !== 'undefined' && (scale > 1 || (scale <= 1 && overflow === 'scroll')) && (
      <ThumbContainer axis={axis} progress={progress} scale={scale}>
        <ThumbSegment dragSignifier axis={axis} position="start" {...thumbStartProps} />
        <ThumbSegment axis={axis} position="middle" {...thumbMiddleProps} />
        <ThumbSegment dragSignifier axis={axis} position="end" {...thumbEndProps} />
      </ThumbContainer>
    )}
  </Track>
);
