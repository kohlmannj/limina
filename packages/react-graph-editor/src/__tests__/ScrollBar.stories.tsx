import { color, number, select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import { ThemeProvider } from 'emotion-theming';
import React from 'react';
import { ScrollBar } from '../components/ScrollBar';
import { defaults } from '../scrollBarTheme';
import { ScrollBarAxis } from '..';

storiesOf('react-graph-editor/ScrollBar', module)
  .add('with knobs', () => {
    const axis = select(
      'Axis',
      { 'Y (Vertical)': 'y', 'X (Horizontal)': 'x' },
      'y',
      'axis'
    ) as ScrollBarAxis;

    return (
      <ThemeProvider
        theme={{
          thumbColor: color('Thumb Color', defaults.thumbColor, 'thumbColor'),
          trackColor: color('Track Color', defaults.trackColor, 'trackColor'),
          thumbWidth: number(
            'Thumb Width',
            defaults.thumbWidth,
            {
              range: true,
              min: 0,
              max: 100,
              step: 0.5,
            },
            'thumbWidth'
          ),
          trackWidth: number(
            'Track Width',
            defaults.trackWidth,
            {
              range: true,
              min: 0,
              max: 100,
              step: 1,
            },
            'trackWidth'
          ),
        }}
      >
        <ScrollBar
          progress={number(
            'Progress',
            0,
            {
              range: true,
              min: -0.5,
              max: 1.5,
              step: 0.01,
            },
            'progress'
          )}
          scale={number(
            'Scale',
            2,
            {
              range: true,
              min: 1,
              max: 10,
              step: 1,
            },
            'scale'
          )}
          axis={axis}
          overflow={select('Overflow', { Auto: 'auto', Scroll: 'scroll' }, 'auto', 'overflow')}
          style={{ [axis === 'y' ? 'height' : 'width']: '400px', margin: '25vh' }}
        />
      </ThemeProvider>
    );
  })
  .add('with default props', () => <ScrollBar style={{ height: '100vh' }} />);

storiesOf('react-graph-editor/ScrollBar/Y-Axis/Scale 2', module)
  .add('with progress 0', () => <ScrollBar scale={2} style={{ height: '100vh' }} />)
  .add('with progress 0.25', () => (
    <ScrollBar progress={0.25} scale={2} style={{ height: '100vh' }} />
  ))
  .add('with progress 0.5', () => (
    <ScrollBar progress={0.5} scale={2} style={{ height: '100vh' }} />
  ))
  .add('with progress 0.75', () => (
    <ScrollBar progress={0.75} scale={2} style={{ height: '100vh' }} />
  ))
  .add('with progress 1', () => <ScrollBar progress={1} scale={2} style={{ height: '100vh' }} />);

storiesOf('react-graph-editor/ScrollBar/Y-Axis/Scale 10', module)
  .add('with progress 0', () => <ScrollBar scale={10} style={{ height: '100vh' }} />)
  .add('with progress 0.25', () => (
    <ScrollBar progress={0.25} scale={10} style={{ height: '100vh' }} />
  ))
  .add('with progress 0.5', () => (
    <ScrollBar progress={0.5} scale={10} style={{ height: '100vh' }} />
  ))
  .add('with progress 0.75', () => (
    <ScrollBar progress={0.75} scale={10} style={{ height: '100vh' }} />
  ))
  .add('with progress 1', () => <ScrollBar progress={1} scale={10} style={{ height: '100vh' }} />);
