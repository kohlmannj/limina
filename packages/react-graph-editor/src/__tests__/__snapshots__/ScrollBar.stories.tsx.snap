// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-graph-editor/ScrollBar with default props 1`] = `
<Component
  axis="y"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <Component
    axis="y"
    scale={1}
  >
    <Component
      axis="y"
      dragSignifier={true}
      position="start"
    />
    <Component
      axis="y"
      position="middle"
    />
    <Component
      axis="y"
      dragSignifier={true}
      position="end"
    />
  </Component>
</Component>
`;

exports[`react-graph-editor/ScrollBar with knobs 1`] = `
<Component
  axis="y"
  overflow="auto"
  progress={0}
  scale={2}
  style={
    Object {
      "height": "400px",
      "margin": "25vh",
    }
  }
/>
`;

exports[`react-graph-editor/ScrollBar/Y-Axis/Scale 2 with progress 0 1`] = `
<Component
  axis="y"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <Component
    axis="y"
    scale={2}
  >
    <Component
      axis="y"
      dragSignifier={true}
      position="start"
    />
    <Component
      axis="y"
      position="middle"
    />
    <Component
      axis="y"
      dragSignifier={true}
      position="end"
    />
  </Component>
</Component>
`;

exports[`react-graph-editor/ScrollBar/Y-Axis/Scale 2 with progress 0.5 1`] = `
<Component
  axis="y"
  progress={0.5}
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <Component
    axis="y"
    progress={0.5}
    scale={2}
  >
    <Component
      axis="y"
      dragSignifier={true}
      position="start"
    />
    <Component
      axis="y"
      position="middle"
    />
    <Component
      axis="y"
      dragSignifier={true}
      position="end"
    />
  </Component>
</Component>
`;

exports[`react-graph-editor/ScrollBar/Y-Axis/Scale 2 with progress 0.25 1`] = `
<Component
  axis="y"
  progress={0.25}
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <Component
    axis="y"
    progress={0.25}
    scale={2}
  >
    <Component
      axis="y"
      dragSignifier={true}
      position="start"
    />
    <Component
      axis="y"
      position="middle"
    />
    <Component
      axis="y"
      dragSignifier={true}
      position="end"
    />
  </Component>
</Component>
`;

exports[`react-graph-editor/ScrollBar/Y-Axis/Scale 2 with progress 0.75 1`] = `
<Component
  axis="y"
  progress={0.75}
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <Component
    axis="y"
    progress={0.75}
    scale={2}
  >
    <Component
      axis="y"
      dragSignifier={true}
      position="start"
    />
    <Component
      axis="y"
      position="middle"
    />
    <Component
      axis="y"
      dragSignifier={true}
      position="end"
    />
  </Component>
</Component>
`;

exports[`react-graph-editor/ScrollBar/Y-Axis/Scale 2 with progress 1 1`] = `
<Component
  axis="y"
  progress={1}
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <Component
    axis="y"
    progress={1}
    scale={2}
  >
    <Component
      axis="y"
      dragSignifier={true}
      position="start"
    />
    <Component
      axis="y"
      position="middle"
    />
    <Component
      axis="y"
      dragSignifier={true}
      position="end"
    />
  </Component>
</Component>
`;

exports[`react-graph-editor/ScrollBar/Y-Axis/Scale 10 with progress 0 1`] = `
<Component
  axis="y"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <Component
    axis="y"
    scale={10}
  >
    <Component
      axis="y"
      dragSignifier={true}
      position="start"
    />
    <Component
      axis="y"
      position="middle"
    />
    <Component
      axis="y"
      dragSignifier={true}
      position="end"
    />
  </Component>
</Component>
`;

exports[`react-graph-editor/ScrollBar/Y-Axis/Scale 10 with progress 0.5 1`] = `
<Component
  axis="y"
  progress={0.5}
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <Component
    axis="y"
    progress={0.5}
    scale={10}
  >
    <Component
      axis="y"
      dragSignifier={true}
      position="start"
    />
    <Component
      axis="y"
      position="middle"
    />
    <Component
      axis="y"
      dragSignifier={true}
      position="end"
    />
  </Component>
</Component>
`;

exports[`react-graph-editor/ScrollBar/Y-Axis/Scale 10 with progress 0.25 1`] = `
<Component
  axis="y"
  progress={0.25}
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <Component
    axis="y"
    progress={0.25}
    scale={10}
  >
    <Component
      axis="y"
      dragSignifier={true}
      position="start"
    />
    <Component
      axis="y"
      position="middle"
    />
    <Component
      axis="y"
      dragSignifier={true}
      position="end"
    />
  </Component>
</Component>
`;

exports[`react-graph-editor/ScrollBar/Y-Axis/Scale 10 with progress 0.75 1`] = `
<Component
  axis="y"
  progress={0.75}
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <Component
    axis="y"
    progress={0.75}
    scale={10}
  >
    <Component
      axis="y"
      dragSignifier={true}
      position="start"
    />
    <Component
      axis="y"
      position="middle"
    />
    <Component
      axis="y"
      dragSignifier={true}
      position="end"
    />
  </Component>
</Component>
`;

exports[`react-graph-editor/ScrollBar/Y-Axis/Scale 10 with progress 1 1`] = `
<Component
  axis="y"
  progress={1}
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <Component
    axis="y"
    progress={1}
    scale={10}
  >
    <Component
      axis="y"
      dragSignifier={true}
      position="start"
    />
    <Component
      axis="y"
      position="middle"
    />
    <Component
      axis="y"
      dragSignifier={true}
      position="end"
    />
  </Component>
</Component>
`;
